src/CMakeLists.txt

```diff
@@ -117,6 +117,16 @@ elseif(${PLATFORM} MATCHES "Raspberry Pi")
   set(PLATFORM_CPP "PLATFORM_RPI")
   set(GRAPHICS "GRAPHICS_API_OPENGL_ES2")
 
+  add_definitions(-D_GNU_SOURCE=1)
```

Comment:
What declarations are missing otherwise?

Reply:
I'm not sure. It had problems finding some standard headers without this line, if I remember correctly.

Reply:
Would be good to know what exactly failed. Might be just a header we are missing.

Reply:
´´´
[  1%] Building C object src/CMakeFiles/raylib.dir/core.c.o
In file included from /opt/vc/include/interface/vcos/vcos.h:118,
                 from /opt/vc/include/interface/vmcs_host/vc_dispmanx.h:33,
                 from /opt/vc/include/EGL/eglplatform.h:110,
                 from /opt/vc/include/EGL/egl.h:36,
                 from /home/pi/node-raylib/node-raylib/vendor/raylib/src/rlgl.h:666,
                 from /home/pi/node-raylib/node-raylib/vendor/raylib/src/core.c:136:
/opt/vc/include/interface/vcos/pthreads/vcos_platform.h: In function ‘vcos_semaphore_wait_timeout’:
/opt/vc/include/interface/vcos/pthreads/vcos_platform.h:307:13: error: implicit declaration of function ‘sem_timedwait’; did you mean ‘sigtimedwait’? [-Werror=implicit-function-declaration]
       ret = sem_timedwait( sem, &ts );
             ^~~~~~~~~~~~~
             sigtimedwait
/home/pi/node-raylib/node-raylib/vendor/raylib/src/core.c: In function ‘EventThread’:
/home/pi/node-raylib/node-raylib/vendor/raylib/src/core.c:5162:13: error: implicit declaration of function ‘usleep’; did you mean ‘sleep’? [-Werror=implicit-function-declaration]
             usleep(5000); // Sleep for 5ms to avoid hogging CPU time
             ^~~~~~
             sleep
```
must have something to do with the VideoCore library. I'll try to research some more.

Reply:
usleep is actually deprecated, I created a separate issue for this.
I have no idea yet what's up with `sem_timedwait`.
Every example - even in the Raspberry Pi userland repo itself - seems really simple. No flags or anything needed, just 2-3 libraries, which I added here.
The Makefile build also runs fine. What's the difference here?

Reply:
I changed it to define `_DEFAULT_SOURCE` instead. That also works, and I got it from the Makefile: https://github.com/raysan5/raylib/blob/72443f8f25b1f8ded3b05da17857b7eeb1556b35/src/Makefile#L275

I hope that settles it for now. Maybe we can improve this situation in the future.